File: 백준/Silver/1920. 수 찾기/수 찾기.py
N=int(input())
A=list(map(int,input().split()))
A.sort()
M=int(input())
B=list(map(int,input().split()))


def bin(key,A,start,end):
    if start>end:
        print(0)
    else:
        mid=(start+end)//2

        if key<A[mid]:
            end=mid-1
            return bin(key, A, start, end)
        elif key>A[mid]:
            start=mid+1
            return bin(key, A, start, end)
        else:
            print(1)
            return


for i in B:
    bin(i,A,0,N-1)

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <stack>
#include <queue>

using namespace std;

typedef pair<int, int> edge;
vector<vector<edge>> vec;
vector<bool> visited;
static void BFS(int start);
vector<int> dist;

int main() {
    ios::sync_with_stdio(false);
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin>>n;
    vector<int> A(n);
    for(int i=0;i<n;i++){
        cin>>A[i];
    }
    sort(A.begin(), A.end());

    int m;
    cin>>m;
    for(int i=0;i<m;i++) {
        int val;
        cin >> val;
        int start=0;
        int end=n-1;
        int status=0;
        while(start<=end){
            int mid=(start+end)/2;
            if(A[mid]>val){
                end=mid-1;
            }
            else if (A[mid]<val){
                start=mid+1;
            }
            else{
                status=1;
                break;
            }
        }
        cout<<status<<'\n';

    }

    }

